### PR TITLE
introducing the HOPL 2017 blog post series

### PR DESCRIPTION
**Pull request by bennn**
- introductory / outline post
- outline for Tony's presentation
- full teaser for Ben's presentation
- outline for Dan's presentation, plus an anecdote from lunch

@dbp @tonyg can you add some text introducing your talks about pointing to the full notes in the repo? (Either make a commit [on a new branch I guess], or just paste the text in a comment below.)

### CODE REVIEW

File: blog/_src/posts/2017-02-15-hopl2017-linear-types.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+TODO
+
+<!-- more -->
+
+TODO
+
+Unofficial notes: [https://github.com/nuprl/hopl-s2017/blob/master/lecture_notes/2017-02-14.md]()
+
+Just after the talk, over lunch, we had a lab discussion about the phrase
```
Comment (bennn) [inline blog/_src/posts/2017-02-15-hopl2017-linear-types.md]: I thought this was a fun discussion, but the prose here is very raw.

Up to you if you even want to keep this section --- we can also put it at the end of the `lecture_notes/`.

Comment (dbp) [inline blog/_src/posts/2017-02-15-hopl2017-linear-types.md]: I like the discussion.

**Comment by dbp** [comment]
We should probably include our lists of papers? I would even say I'd like to have my annotated bibliography. Matthias said we was posting it... somewhere? I don't know where though.

**Comment by bennn** [comment]
somewhere = in a folder in the HOPL repo. He put mine in the "datalog" folder I'd made, and Ming-Ho's in a new folder.

(I don't think I want to change my blog post, but I'll mention the papers + bibs in the intro post.)

**Comment by dbp** [comment]
My teaser:

In this talk, we covered early papers (primarily, by Girard, Lafont, and Abramsky) on linear logic and its reflections into computation. The goal was to understand why linearity is often turned to as a principled way to control resource usage, as shows up in a language like Rust. From the very beginning, researchers realized the implications for "low-level" languages - that linear resources would eliminate the need for garbage collection, allow in-place mutation, and enable safe parallel computation. However, pure implementations of linearity incur lots of copying, doing away with any efficiency gained, and we covered a survey of papers that attempted to reconcile this contradiction by weakening linearity in controlled ways.

**Comment by tonyg** [comment]
When programs are written with concurrency in mind, the programmer reasons about the interactions between concurrent components or agents in the program. This includes exchange of information, as well as management of resources, handling of partial failure, collective decision-making and so on.

These components might be objects, or threads, or processes, or actors, or some more nebulous and loosely-defined concept; a group of callbacks, perhaps. The programmer has the notion of an agent in their mind, which translates into some representation of that agent in the program.

We think about the contexts (because there can be more than one) in which agents exist in two different ways. From each agent's perspective, the important thing to think about is the boundary between the agent and everything else in the system. But from the system perspective, we often think about *conversations* between agents, whether it's just two having an exchange, or a whole group collaborating on some task. Agents in a conversation play different roles, join and leave the group, and build shared conversational state.

In this talk, I used the idea of these *conversational contexts* as a lens through which to view the development of various metaphors and mechanisms of communication and coordination. I presented four *computational models* for concurrent interaction:

 - monitors, and shared memory concurrency generally
 - the actor model
 - channel-based communication
 - tuplespaces

These aren't full programming languages, but there are many *programming models* that build upon them. In some cases, development of these ideas has progressed all the way up to *system models* including user interaction and so forth.

The linked lecture notes include informal sketches of reduction semantics for each of the four models, plus a handful of small examples to give a feel for them.

**Comment by bennn** [comment]
Great! I'll put these online tonight.

File: blog/_src/posts/2017-02-15-hopl2017-intro.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+    Title: Introducing HOPL 2017
+    Date: 2017-02-15T01:21:37
+    Tags: HOPL
```
Comment (gasche) [inline blog/_src/posts/2017-02-15-hopl2017-intro.md]: The author tag seems to be missing.

**Comment by gasche** [comment]
From a purely editorial point of view, I wonder if it wouldn't be interesting to not release all three blog posts at the same time. Would it make sense to send only one post each day, starting with the general one, and talk-specific posts in talk order?

**Comment by bennn** [comment]
I created the posts in talk-order, so they appear in the right order when I build the blog locally.
(The order's the only reason we'd want to delay posting, right?)

**Comment by mhyee** [comment]
As Twitter Tsar, I like @gasche's idea to space out the posts. It seems more effective than posting all three at the same time.

We could even make it a regular schedule. Every Tuesday and Friday, we put out a new post, lagging behind the actual talk by, e.g. 2 weeks.

**Comment by gasche** [comment]
My idea is that the people that subscribe to the blog post have a limited blog-reading time budget each day, and tend to not go back to days-old unread blog posts to read them. By spacing the posts out, you increase the chance that they are able to read them.

**Comment by bennn** [comment]
Ok, I added `DRAFT` tags to the teasers. When we merge this PR, only the intro post will be public.

**Comment by bennn** [comment]
Okay, these are merged & the intro post should be live soon.

Planning to publish Tony's tomorrow, mine on Tuesday, and Dan's on next Tuesday (to get us on the 2-week delay)

**Pull request closed without merging**